### DISCUSSION TITLE
Webpack installation failure

### DISCUSSION THREAD

**Discussion by Tsjippy** [webpack/webpack#15065] [Q&A] [answered]
# Bug report

**What is the current behavior?**
Webpack seems to install correctly but is not.

**If the current behavior is a bug, please provide the steps to reproduce.**

- Install nodejs
- create new folder
- run  `npm init -y`
- run `npm install webpack webpack-cli --save-dev`
Result: 
```
up to date, audited 1 package in 20s
found 0 vulnerabilities
```

- Check the node_modules folder:
```
d-----        28-12-2021     11:08                @discoveryjs
d-----        28-12-2021     11:08                @types
d-----        28-12-2021     11:08                @webassemblyjs
d-----        28-12-2021     11:08                @webpack-cli
d-----        28-12-2021     11:08                @xtuc
-a----        28-12-2021     11:08            111 .package-lock.json
```

- Run webpack: `webpack : The term 'webpack' is not recognized as the name of a cmdlet, function, script file, or operable program.`
- webpack is not found in the folder nor in %AppData%\npm\node_modules nor in C:\Program Files\nodejs\node_modules

- run `npm update`:
```
added 119 packages, and audited 120 packages in 2m
15 packages are looking for funding
  run `npm fund` for details
found 0 vulnerabilities
```

- Everything is installed correctly

**What is the expected behavior?**
I should not have to invoke npm update, the first install should work straight away

**Other relevant information:**
System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 8.60 GB / 15.74 GB
  Binaries:
    Node: 14.17.6 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 96.0.4664.110
    Edge: Spartan (44.22000.120.0), Chromium (95.0.1020.40)
    Internet Explorer: 11.0.22000.120
  Packages:
    webpack: ^5.65.0 => 5.65.0
    webpack-cli: ^4.9.1 => 4.9.1

**Comment by alexander-akait** [comment]
I think you have something wrong with your environment, what is npm version?

**Comment by Tsjippy** [comment]
Node: 14.17.6 - C:\Program Files\nodejs\node.EXE
npm: 8.3.0 - C:\Program Files\nodejs\npm.CMD

I just found out that if I install a new package with `npm install --save-dev` I have to run `npm update` again.

I have tried to reset my environment by uninstalling webpack and nodejs and reinstalling, same result

**Comment by alexander-akait** [comment]
Can you run `npm ls` after installation?

**Comment by Tsjippy** [comment]
ok, here is the result:
```
npm install webpack webpack-cli --save-dev
up to date, audited 1 package in 22s
found 0 vulnerabilities
```


PS D:\Nieuwe map> npm ls
```
nieuwe-map@1.0.0 D:\Nieuwe map
+-- UNMET DEPENDENCY webpack-cli@^4.9.1
`-- UNMET DEPENDENCY webpack@^5.65.0

npm ERR! code ELSPROBLEMS
npm ERR! missing: webpack-cli@^4.9.1, required by nieuwe-map@1.0.0
npm ERR! missing: webpack@^5.65.0, required by nieuwe-map@1.0.0

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ewald\AppData\Local\npm-cache\_logs\2021-12-28T12_05_08_872Z-debug.log
```

**Reply by alexander-akait** [reply]
It means `npm install webpack webpack-cli --save-dev` doesn't add packages to `package.json`, can you check your permissions of file?

**Reply by Tsjippy** [reply]
package json after install:
```
{
  "name": "nieuwe-map",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "webpack": "^5.65.0",
    "webpack-cli": "^4.9.1"
  }
}
```

**Comment by Tsjippy** [comment]
I feel like npm thinks the packages are already installed somewhere but they are not

**Reply by alexander-akait** [reply]
Maybe you have multiple npm versions? Can you check it?

**Reply by Tsjippy** [reply]
npm -v gives me: 8.1.2

I have a C:\Program Files\nodejs\node_modules\npm folder
I also have a %appdata%\npm folder

**Reply by alexander-akait** [reply]
You can open an issue in npm in this case, sorry, it is out of scope webpack

**Reply by Tsjippy** [reply]
Ok thank you

**Comment by Tsjippy** [comment, marked as answer]
I finally fixed it by completely reinstalling my os
